Compilation without threads does not seem to work

On master and release/6.3.0, the following configure command
```
./configure --enable-static --disable-shared --disable-examples --disable-flatzinc --disable-gist --disable-qt --disable-float-vars --disable-thread --disable-doc-dot --disable-profile --disable-debug -disable-doc-tagfile
```
leads to the following subsequent compilation error
```
$ make
make[1]: `gecode/kernel/var-type.hpp' is up to date.
make[1]: `gecode/kernel/var-imp.hpp' is up to date.
g++ -I. -Qunused-arguments -fno-rounding-math -ffinite-math-only -fno-math-errno -fno-strict-aliasing -O3 -fvisibility=hidden -ggdb -std=c++17 -pipe -Wno-unknown-pragmas -Wall -Wextra -fPIC -DNDEBUG   \
	-c -o gecode/support/exception.o  gecode/support/exception.cpp
In file included from gecode/support/exception.cpp:34:
In file included from ./gecode/support.hh:102:
./gecode/support/thread.hpp:114:2: error: No suitable mutex implementation found
#error No suitable mutex implementation found
 ^
1 error generated.
make[1]: *** [gecode/support/exception.o] Error 1
make: *** [compilelib] Error 2
```

Tested on macOS. 

Reported in https://groups.google.com/g/gecode/c/UzjhYx0msYw

--- a/changelog.in
+++ b/changelog.in
@@ -70,6 +70,15 @@ Date: 2020-??-??
 Let's see.
 
 [ENTRY]
+Module: other
+What:   bug
+Rank:   minor
+Thanks: Joerg Strebel
+Issue:  136
+[DESCRIPTION]
+Fix compilation without threads
+
+[ENTRY]
 Module: gist
 What:   change
 Rank:   minor

--- a/gecode/search/bab.cpp
+++ b/gecode/search/bab.cpp
@@ -43,8 +43,8 @@ namespace Gecode { namespace Search {
 
   Engine*
   babengine(Space* s, const Options& o) {
-#ifdef GECODE_HAS_THREADS
     Options to = o.expand();
+#ifdef GECODE_HAS_THREADS
     if (to.threads == 1.0) {
       if (to.tracer)
         return new WorkerToEngine<Seq::BAB<TraceRecorder>>(s,to);

--- a/gecode/search/dfs.cpp
+++ b/gecode/search/dfs.cpp
@@ -43,8 +43,8 @@ namespace Gecode { namespace Search {
 
   Engine*
   dfsengine(Space* s, const Options& o) {
-#ifdef GECODE_HAS_THREADS
     Options to = o.expand();
+#ifdef GECODE_HAS_THREADS
     if (to.threads == 1.0) {
       if (to.tracer)
         return new WorkerToEngine<Seq::DFS<TraceRecorder>>(s,to);

--- a/gecode/support/thread.hpp
+++ b/gecode/support/thread.hpp
@@ -110,8 +110,10 @@ namespace Gecode { namespace Support {
 #elif defined(GECODE_THREADS_PTHREADS)
     /// The Pthread mutex
     pthread_mutex_t p_m;
-#else
+#elif defined(GECODE_HAS_THREADS)
 #error No suitable mutex implementation found
+#else
+    // When no threads are used, Mutex is a no-op
 #endif
   public:
     /// Initialize mutex

